prometheus/collect_topic_partition_offsets.go

```diff
@@ -101,5 +101,56 @@ func (e *Exporter) collectTopicPartitionOffsets(ctx context.Context, ch chan<- p
 		}
 	}
 
+	// Highest Timestamp Offsets
+	// NB: this requires Kafka Brokers 3.0+ (see https://issues.apache.org/jira/browse/KAFKA-12541)
+	// In older versions this is returning the timestamp of the low watermarks (earliest offset)
+	maxTimestampOffsets, err := e.minionSvc.ListOffsetsCached(ctx, -3)
+	if err != nil {
+		e.logger.Warn("failed to fetch offsets for max timestamp", zap.Error(err))
+		//return false
```

Comment:
any reasons we don't want returning false here and following same pattern as above?

Reply:
TODO: good point - initially I thought -3 would create issues in older versions but it's not so I'll revert that to stricter check